addons/hr_holidays/models/hr.py

```diff
@@ -0,0 +1,216 @@
+                    "to manage the leave days of the employees if the configuration does not allow to use this field.") % (len(status_list)))
+            status = status_list[0] if status_list else None
+            if not status:
+                continue
+            # if a status is found, then compute remaing leave for current employee
+            difference = employee.remaining_leaves - actual_remaining.get(employee.id, 0)
+            if difference > 0:
+                leave = self.env['hr.holidays'].create({
+                    'name': _('Allocation for %s') % employee.name,
+                    'employee_id': employee.id,
+                    'holiday_status_id': status.id,
+                    'type': 'add',
+                    'holiday_type': 'employee',
+                    'number_of_days_temp': difference
+                })
+                for signal in ('confirm', 'validate', 'second_validate'):
+                    leave.signal_workflow(signal)
+            elif difference < 0:
+                raise UserError(_('You cannot reduce validated allocation requests'))
+
+    @api.multi
+    def _compute_leave_status(self):
+        # Used SUPERUSER_ID to forcefully get status of other user's leave, to bypass record rule
+        holidays = self.env['hr.holidays'].sudo().search([
+            ('employee_id', 'in', self.ids),
+            ('date_from', '<=', fields.Datetime.now()),
+            ('date_to', '>=', fields.Datetime.now()),
+            ('type', '=', 'remove'),
+            ('state', 'not in', ('cancel', 'refuse'))
+        ])
+        leave_data = {}
+        for holiday in holidays:
+            leave_data[holiday.employee_id.id] = {}
+            leave_data[holiday.employee_id.id]['leave_date_from'] = holiday.date_from
+            leave_data[holiday.employee_id.id]['leave_date_to'] = holiday.date_to
+            leave_data[holiday.employee_id.id]['current_leave_state'] = holiday.state
+            leave_data[holiday.employee_id.id]['current_leave_id'] = holiday.holiday_status_id.id
+
+        for employee in self:
+            employee.leave_date_from = leave_data.get(employee.id, {}).get('leave_date_from')
+            employee.leave_date_to = leave_data.get(employee.id, {}).get('leave_date_to')
+            employee.current_leave_state = leave_data.get(employee.id, {}).get('current_leave_state')
+            employee.current_leave_id = leave_data.get(employee.id, {}).get('current_leave_id')
+
+    @api.multi
+    def _compute_leaves_count(self):
+        leaves = self.env['hr.holidays'].read_group([
+            ('employee_id', 'in', self.ids),
+            ('holiday_status_id.limit', '=', False),
+            ('state', '=', 'validate')
+        ], fields=['number_of_days', 'employee_id'], groupby=['employee_id'])
+        mapping = dict([(leave['employee_id'][0], leave['number_of_days']) for leave in leaves])
```

Comment:
use a dictionary comprehension, or at least remove the redundant list comprehension